### PR TITLE
docs: add migration guide for v7

### LINKED ISSUE #1230: Effects: Write migration from V6 to V7 to handle compatibility changes
## If accepted, I would be willing to submit a PR for this feature

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

With https://github.com/ngrx/platform/issues/860 and the removal of instance operators, we'll need a way to migrate users usage of the existing `Actions` from `@ngrx/effects` to `Actions` from `@ngrx/effects/compat` and to the pipeable `ofType` operator where necessary.

This also covers migrating from `store.select()` to `store.pipe(select())`

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1230 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Closes #1230

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -508,14 +596,12 @@ import * as RouterActions from './actions/router';
...
   @Effect({ dispatch: false })
-  navigate$ = this.actions$
-    .ofType(RouterActions.GO)
-    .pipe(
-      map((action: RouterActions.Go) => action.payload),
-      tap(({ path, query: queryParams, extras }) =>
-        this.router.navigate(path, { queryParams, ...extras })
-      )
-    );
+  navigate$ = this.actions$.ofType(RouterActions.GO).pipe(
```
Comment (diicar) [inline MIGRATION.md]: this.actions$.ofType(RouterActions.GO).pipe(map...  >>>  this.actions$.pipe(ofType(RouterActions.GO), map(() => ...))

Comment (timdeschryver) [reply MIGRATION.md]: I don't think this is necessary since these are the v4 docs.

**Comment by timdeschryver** [comment]
Added more breaking changes - see https://github.com/ngrx/platform/issues?utf8=%E2%9C%93&q=label%3A%22Breaking+Change%22++label%3A7.x++ for ref

File: MIGRATION.md [hunk 2]
```diff
@@ -4,10 +4,10 @@
 
 Links to the current documentation for ngrx 4.x
 
-* [@ngrx/store](./docs/store/README.md)
```
Comment (brandonroberts) [inline MIGRATION.md]: Revert the changes to this file

Comment (timdeschryver) [reply MIGRATION.md]: Done.

File: projects/ngrx.io/content/guide/migration/v7.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  someEffect$: Observable<Action> = this.actions$.pipe(
+    ofType(UserActions.LOGIN), // use the pipeable ofType operator
+    map(() => new AnotherAction())
+  );
+
+  constructor(private actions$: Actions) {}
+}
+```
+
+## @ngrx/store-router
```
Comment (kouMatsumoto) [inline projects/ngrx.io/content/guide/migration/v7.md]: `## @ngrx/store-router`

This should be `@ngrx/router-store` instead of `@ngrx/store-router`, shouldn't it?

Comment (timdeschryver) [reply projects/ngrx.io/content/guide/migration/v7.md]: Yea, you're right 😅

File: projects/ngrx.io/content/guide/migration/v7.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+```ts
+{type: '@ngrx/store/update-reducers', features: ['feat1', 'feat2']}
+```
+
+## @ngrx/effects
+
+### `ofType`
+
+In NgRx 6.1 the `ofType` static function was marked as deprecated in favor of the `ofType` RxJS operator, in NgRx v7 this static function was dropped.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v7.md]: `ofType` RxJS operator -> pipeable `ofType` operator

Comment (brandonroberts) [reply projects/ngrx.io/content/guide/migration/v7.md]: Remove `static`

**Comment by brandonroberts** [comment]
We also need to add a section to the `projects/ngrx.io/content/navigation.json` for migrations

File: projects/ngrx.io/content/guide/migration/v7.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+# Migration guide v7
+
+## @ngrx/store
```
[the remark this answers is not in the document]
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/migration/v7.md]: I'm going to keep the section but remove `select`. Because we have also have `ngrx/store/update-reducers`. Would that be OK for you?

**Comment by timdeschryver** [comment]
The migration pages looks as follows:

image

File: projects/ngrx.io/content/navigation.json [hunk 6]
```diff
@@ -8,6 +8,10 @@
       "url": "docs",
       "title": "Docs"
     },
+    {
+      "url": "guide/migration/v7",
```
Comment (brandonroberts) [inline projects/ngrx.io/content/navigation.json]: Let's make this a side nav section called `Migrations` with a `V7` menu item under it.

Comment (timdeschryver) [reply projects/ngrx.io/content/navigation.json]: Something like this? 
image
